tests/operations.go

```diff
@@ -309,3 +323,45 @@ func (ops *Operations) ExecPod(opts *Options) ([]byte, error) {
 	Expect(execOut.Len()).Should(BeNumerically(">", 0), "while streaming the command in pod ", opts.podName, execErr.String(), execOut.String())
 	return execOut.Bytes(), nil
 }
+
+// GetPodCompletedCountEventually gives the number of pods running eventually
+func (ops *Operations) GetPodCompletedCountEventually(namespace, lselector string, expectedPodCount int) int {
+	var podCount int
+	for i := 0; i < maxRetry; i++ {
+		podCount = ops.GetPodCompletedCount(namespace, lselector)
+		if podCount == expectedPodCount {
+			return podCount
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return podCount
+}
+
+// GetPodCompletedCount gives number of pods running currently
+func (ops *Operations) GetPodCompletedCount(namespace, lselector string) int {
+	pods, err := ops.PodClient.
+		WithNamespace(namespace).
+		List(metav1.ListOptions{LabelSelector: lselector})
+	Expect(err).ShouldNot(HaveOccurred())
+	return pod.
+		ListBuilderForAPIList(pods).
+		WithFilter(pod.IsCompleted()).
+		List().
+		Len()
+}
+
+// CheckUpgradeResult checks whether all the tasks in upgraderesult
+// have success
+func (ops *Operations) CheckUpgradeResult(namespace, lselector string) bool {
+	urList, err := ops.URClient.
+		WithNamespace(namespace).
+		List(metav1.ListOptions{LabelSelector: lselector})
+	Expect(err).ShouldNot(HaveOccurred())
+	for _, task := range urList.Items[0].Tasks {
+		if task.Status == "Fail" {
```

Comment:
Logic should check if `Status != "Success"`
Name of the method & behaviour should reflect each other.

Reply:
In case there is no snapshot of the volume then the runtask will skip last two tasks and the status will come as empty string. If we check for Success then it will return false.

Reply:
Ok then Let us rename this method to `VerifyUpgradeResultTasksIsNotFail`